pthread/pthread.c

```diff
@@ -201,8 +258,22 @@ void pthread_exit(void *value_ptr)
 {
 	pthread_ctx *ctx = (pthread_ctx *)pthread_self();
 
-	if (ctx != NULL)
+	if (ctx != NULL) {
 		ctx->retval = value_ptr;
+		mutexLock(pthread_common.pthread_key_lock);
+		while (ctx->key_data_list != NULL) {
+			pthread_key_data_t *thread_data = ctx->key_data_list;
+			void *value = thread_data->value;
+			void (*destructor)(void *) = thread_data->key->destructor;
```

Comment:
Our style is still variable declaration on the beginning of the function, not beginning of the block. This might change, but for now it is